### ISSUE TITLE
Suspended tabs' favicons are stored correctly but never refreshed/shown unless manually clicked

### ISSUE THREAD

**Issue by CollinChaffin** [deanoemcke/thegreatsuspender#784]
Latest commit `bf02517` (Oct 8th) has another issue where suspended tabs' favicons are stored correctly but never refreshed/updated after suspended until each is manually clicked.  Note - this is not to activate and un-suspend, but only to click the physical tab on the bar and it is only at that time does the stored icon/favicon for that suspended tab suddenly refreshed and shown.

Here is a screencap below demonstrating this bug (the cursor was captured as a smaller crosshair/dot and is what is clicking each of the suspended tabs to demonstrate the bug).  Also, that last tab click is very fast since the video loops so you have to watch carefully to see it also change - sorry!

![tgs_10-09-2018-favicons](https://user-images.githubusercontent.com/9301483/46687128-b32c0a00-cbbf-11e8-871f-19d3e67ce1fe.gif)

**Comment by CollinChaffin** [comment]
Yes could be duplicate - but no, not fixed with latest commit 238bfa1.  I saved a number of tabs that were suspended as I always used to in my session manager (Session Buddy) that simply saves each tab with it's URL (in case of suspended it is the suspended chrome:// extension of TGS).....restores fine.  The issue is somewhere in the last couple weeks the refresh code changed and now, they all have no icons and just the generic blockheads until each tab is manually clicked upon as shown in my post above.

I do want to reiterate that this all worked 100% fine a month ago back years and only broke recently.  The same goes for the other couple open issues I have right now they all began in this latest round of code improvements.

I'm still testing the other issues to see if those are resolved.  I have not diffed the code from a month ago to even begin trying to debug how it was refreshing the tabs on initial load/refresh vs today - but I will here over the next day or so to try to help.

**Comment by CollinChaffin** [comment]
I want to add that not only is this an issue, but just plain old "wait 15 min and watch a tab suspend" results in that tab's appearance now also matching what I've stated above - a plain blockhead icon tab with no proper favicon that usually is set upon the suspend process refreshing the tab title/icon to match what it was when live.  So it is not only in that loading/refresh process not showing the favicon anymore, but also not in the case of a normal simple suspend->set/refresh title->set/refresh favicon based on whatever suspend timeout the user has chosen.

**Comment by deanoemcke** [comment]
I'm hope you're not actually waiting for 15 minutes to watch a tab suspend. There's a 20 second timer that might save you a bit of time if you are ;)
I don't think that the issue is caused by new code. It seems more likely that it's a new version of chrome that is causing the issue.
Like I've mentioned elsewhere, I can replicate this issue if I have a window full of suspended tabs, and then close the whole window, and then use the 'undo closed tab' option which restores the whole window. When I do this, most of the suspended tabs fail to restore with their proper icons. I haven't seen this happen with normal tab suspension though, although it seems likely that it's a similar thing happening with chrome.
I'll see if I can try to replicate your issue somehow. Can you please give me your operating system version, your chrome version, and the latest version of the extension that you've tried?

**Comment by deanoemcke** [comment]
I have made another optimistic fix for this issue, as I was able to sometimes recreate it, but only with the new 'Apply Chrome's built-in memory-saving when suspending' checked and only after immediately loading a window with 100+ suspended tabs.

So I'm not sure I'm targeting the same issue as you have reported, but the symptoms are at least the same.

Here is the relevant commit: https://github.com/deanoemcke/thegreatsuspender/commit/c1fcbbf757a735117ce65122f918480e267c6104

I will close this issue as a duplicate of #478.
